File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, std::size_t> unique_nums;

        for (std::size_t idx = 0; idx < nums.size(); ++idx) {
            auto it = unique_nums.find(target - nums[idx]);
            if (it != unique_nums.end()) {
                return {it->second, idx};
            }
            unique_nums[nums[idx]] = idx;
        }
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        s, e = 0, len(nums) - 1

        while s < e:
            diff = (e - s) // 2
            mid = s + diff

            if nums[mid] > nums[e]:
                s = mid + 1
            else:
                e = mid

        return nums[s]